src/core/tests/Test_UbuntuProClient.py

```diff
@@ -0,0 +1,195 @@
+            mock_method = getattr(self, method_name)
+            setattr(sys.modules['uaclient.api.u.pro.security.status.reboot_required.v1'], mock_name, mock_method)
+        else:
+            reboot_module = imp.new_module('reboot_module')
+            mock_method = getattr(self, method_name)
+            setattr(reboot_module, mock_name, mock_method)
+            sys.modules['uaclient'] = reboot_module
+            sys.modules['uaclient.api'] = reboot_module
+            sys.modules['uaclient.api.u'] = reboot_module
+            sys.modules['uaclient.api.u.pro'] = reboot_module
+            sys.modules['uaclient.api.u.pro.security'] = reboot_module
+            sys.modules['uaclient.api.u.pro.security.status'] = reboot_module
+            sys.modules['uaclient.api.u.pro.security.status.reboot_required'] = reboot_module
+            sys.modules['uaclient.api.u.pro.security.status.reboot_required.v1'] = reboot_module
+
+    def mock_unimport_uaclient_reboot_required_module(self):
+        if sys.version_info[0] == 3:
+            sys.modules['uaclient.api.u.pro.security.status.reboot_required.v1'] = ''
+        else:
+            sys.modules['uaclient'] = ''
+            sys.modules['uaclient.api'] = ''
+            sys.modules['uaclient.api.u'] = ''
+            sys.modules['uaclient.api.u.pro'] = ''
+            sys.modules['uaclient.api.u.pro.security'] = ''
+            sys.modules['uaclient.api.u.pro.security.status'] = ''
+            sys.modules['uaclient.api.u.pro.security.status.reboot_required'] = ''
+            sys.modules['uaclient.api.u.pro.security.status.reboot_required.v1'] = ''
+
+class TestUbuntuProClient(unittest.TestCase):
+    def setUp(self):
+        self.runtime = RuntimeCompositor(ArgumentComposer().get_composed_arguments(), True, Constants.APT)
+        self.container = self.runtime.container
+
+    def tearDown(self):
+        self.runtime.stop()
+
+    def mock_run_command_output_raise_exception(self):
+        raise
+
+
+    def test_install_or_update_pro_success(self):
+        package_manager = self.container.get('package_manager')
+        self.assertIsNone(package_manager.ubuntu_pro_client.install_or_update_pro())
+
+    def test_install_or_update_pro_failure(self):
+        self.runtime.set_legacy_test_type('SadPath')
+        package_manager = self.container.get('package_manager')
+        self.assertIsNone(package_manager.ubuntu_pro_client.install_or_update_pro())
+
+    def test_install_or_update_pro_exception(self):
+        package_manager = self.container.get('package_manager')
+        package_manager.env_layer.run_command_output = self.mock_run_command_output_raise_exception
+        self.assertIsNone(package_manager.ubuntu_pro_client.install_or_update_pro())
+
+    def test_is_pro_working_success(self):
+        obj = MockVersionResult()
+        obj.mock_import_uaclient_version_module('version', 'mock_version')
+        #self.mock_import_uaclient_version_module('version', 'mock_version')
```

Comment:
code commented out

Reply:
Minor: Same thing about commenter resolving after verifying change made its way into code. Unresolving as the iteration with the change is probably not pushed.